wandb/sdk/internal/internal_api.py

```diff
@@ -3513,6 +3522,59 @@ def replace(a: str, b: str) -> None:
         link_artifact: Dict[str, Any] = response["linkArtifact"]
         return link_artifact
 
+    def _resolve_org_entity_name(
+        self, entity: str, organization: str, project: str, portfolio_name: str
+    ) -> str:
+        org_fields = self.server_organization_introspection()
+        can_fetch_org_entity = "orgEntity" in org_fields
+        if not organization and not can_fetch_org_entity:
+            raise ValueError(
+                """Fetching Registry artifacts without inputting an organization is unavailable for your server version.
+                Please upgrade your server to XX or later."""
+            )
+        elif can_fetch_org_entity:
+            org_entity, org_name = self.fetch_org_entity_from_entity(entity)
+            if organization:
+                if organization != org_name and organization != org_entity:
+                    raise ValueError(
+                        f"Input wrong organization: {organization} for registry: {project}/{portfolio_name}"
+                    )
+        else:
+            # Server doesn't support fetching org entity, so we assume the org entity is correctly inputted
+            org_entity = organization
+
+    def fetch_org_entity_from_entity(self, entity: str) -> Tuple[str, str]:
+        query = gql(
+            """
+            query FetchOrgEntityFromEntity(
+                $entityName: String!,
+            ) {
+                entity(name: $entityName) {
+                    organization {
+                        name
+                        orgEntity {
+                            name
+                        }
+                    }
+                }
+            }
+            """
+        )
+        response = self.gql(
+            query,
+            variable_values={
+                "entityName": entity,
+            },
+        )
+        try:
+            org = response["organization"]
+            org_name = org["name"] or ""
+            org_entity_name = org["orgEntity"]["name"] or ""
+        except (LookupError, TypeError) as e:
+            raise ValueError(f"Organization not found for entity {entity!r}") from e
```

Comment:
```suggestion
            raise ValueError(f"Unable to find organization for artifact under entity: {entity!r} Please make sure you are using a team entity when linking to the Registry") from e
```

Should we make the errors consistent between the internal_api and core-service?